core/tracing/src/main/scala/org/typelevel/otel4s/trace/Tracer.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+package trace
+
+import cats.effect.kernel.Resource
+import org.typelevel.otel4s.meta.InstrumentMeta
+
+trait Tracer[F[_]] {
+
+  /** Instrument metadata. Indicates whether instrumentation is enabled or not.
+    */
+  def meta: Tracer.Meta[F]
+
+  /** Returns the context of a span when it is available in the scope.
+    */
+  def currentSpanContext: F[Option[SpanContext]]
+
+  /** Creates a new [[SpanBuilder]]. The builder can be used to make a fully
+    * customized [[Span]].
+    *
+    * @param name
+    *   the name of the span
+    */
+  def spanBuilder(name: String): SpanBuilder[F]
+
+  /** Creates a new tracing scope with a custom parent. A newly created non-root
+    * span will be a child of the given `parent`.
+    *
+    * @example
+    *   {{{
+    * val tracer: Tracer[F] = ???
+    * val span: Span[F] = ???
+    * val customChild: Resource[F, Span[F]] =
```

Comment:
Indentation got funky here

Reply:
Hm, we use this indentation for example blocks all over the project. We can switch to 3 whitespaces after the asterisk.

Reply:
Hmm, you're right.  I don't know why this one jumped out at me.  The `{{{` being three spaces from the code looks weird, but if it's that way everywhere, and it works, it's fine by me.